### PR TITLE
Fix from_iso8601_timestamp_nanos examples

### PR DESCRIPTION
**Pull request by mosabua**
Fixes https://github.com/trinodb/trino/issues/7658

### CODE REVIEW

File: docs/src/main/sphinx/functions/datetime.rst [hunk 1]
```diff
@@ -97,13 +97,13 @@ Date and time functions
     zone, into a ``timestamp(9) with time zone``. The time defaults to
     ``00:00:00.000000000``, and the time zone defaults to the session time zone::
 
-        SELECT from_iso8601_timestamp('2020-05-11');
+        SELECT from_iso8601_timestamp_nanos('2020-05-11');
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/datetime.rst:100]: This does not actually work despite the docs about the default to zero above. I get 

Text '2020-05-11' could not be parsed at index 10

Comment (Ordinant) [reply docs/src/main/sphinx/functions/datetime.rst:100]: Using cli 357 connected trino 357, from_iso8601_timestamp_nanos() is not recognized. Is it supposed to be only in SEP?

Comment (mosabua) [reply docs/src/main/sphinx/functions/datetime.rst:100]: I tested this locally with Trino 356.

**Review by jirassimok — APPROVED** [review]
Looks good for docs; we will need to figure out separately what to do with the buggy behavior of the function itself.

**Comment by hashhar** [comment]
@jirassimok Can you create an issue with your findings of the inconsistent/buggy behavior so that we don't forget?

**Comment by jirassimok** [comment]
Issue opened as #8124.

**Comment by mosabua** [comment]
I updated this PR to document the current behaviour based on discussion with @electrum and the linked issue we created. I suggest we merge this to fix the currently wrong docs and then update when the implementation is done.

Related .. do we need a from_iso8601_timestamp_picos ?

**Comment by jirassimok** [comment]
Writing `from_iso8601_timestamp_picos` would require either a trick like the one I described [here](https://github.com/trinodb/trino/issues/8124#issuecomment-850713263) or we'd have to write our own ISO 8601 parser; I don't think it's a good idea.

**Comment by mosabua** [comment]
@martint @electrum can we get this in before the release..

**Review by Ordinant — APPROVED** [review]
Tested both examples with 356-e.1 and got the results shown.

File: docs/src/main/sphinx/functions/datetime.rst [hunk 2]
```diff
@@ -93,17 +93,13 @@ Date and time functions
 
 .. function:: from_iso8601_timestamp_nanos(string) -> timestamp(9) with time zone
 
-    Parses the ISO 8601 formatted date ``string``, optionally with time and time
-    zone, into a ``timestamp(9) with time zone``. The time defaults to
-    ``00:00:00.000000000``, and the time zone defaults to the session time zone::
-
-        SELECT from_iso8601_timestamp('2020-05-11');
-        -- 2020-05-11 00:00:00.000000000 America/Vancouver
+    Parses the ISO 8601 formatted date and time ``string``.The time zone
```
Comment (m57lyra) [inline docs/src/main/sphinx/functions/datetime.rst:96]: Missing a space after period

Comment (mosabua) [reply docs/src/main/sphinx/functions/datetime.rst:96]: fixed

**Review by rosewms — APPROVED** [review]

**Review by electrum — APPROVED** [review]

**Pull request merged**